src/job.js

```diff
@@ -35,6 +35,8 @@ export default class Job extends EventEmitter {
 
   executeOn(thread) {
     thread
+      .off('progress')
```

Comment:
Might be better to remove the progress listener after being done, especially for garbage collection. Could do that using `thread.promise().then(() => thread.off('progress'), () => thread.off('progress'))`.

But the `thread.off('progress')` behavior might cause side effects, since it does not only remove the listeners we set here in the job, but all others as well. `thread.removeListener('progress', this.onProgress)` would be preferable, but we would also need to move `this.emit('progress', e)` to `onProgress` and bind it beforehand.